Composer/packages/client/src/utils/publishStatusPollingUpdater.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import httpClient from './httpUtil';
+
+enum PollingStateEnum {
+  Stopped,
+  Running,
+}
+
+export enum ApiStatus {
+  Publishing = 202,
+  Success = 200,
+  Failed = 500,
+  Unknown = 404,
+  HeaderTooLarge = 431,
+  BadRequest = 400,
+}
+
+interface OnDataPayload {
+  botProjectId: string;
+  targetName: string;
+  apiResponse: any;
+}
+type OnDataHandler = (payload: OnDataPayload) => void;
+
+export class PublishStatusPollingUpdater {
+  private status: PollingStateEnum = PollingStateEnum.Stopped;
+  private timerId = 0;
+  private publishTargetName: string;
+  private botProjectId: string;
+  private pollingInterval = 10000;
+
+  constructor(botProjectId: string, publishTargetId: string) {
+    this.botProjectId = botProjectId;
+    this.publishTargetName = publishTargetId;
+  }
+
+  private async fetchPublishStatusData(botProjectId: string, publishTargetName: string, onData: OnDataHandler) {
+    if (typeof onData !== 'function') {
+      console.error(new Error('onData should be a function.'));
```

Comment:
why not throw this Error here? instead of logging it

Reply:
move the log in start function